tests/CMakeLists.txt

```diff
@@ -30,32 +30,7 @@ target_link_libraries(models_test
 # Get the list of sources from the test target.
 get_target_property(test_sources models_test SOURCES)
 
-# Go through the list of test sources and parse the test suite name.
```

Comment:
Should we really remove this?  If we transitioned to using catch, then I think we can regex on `TEST_CASE(..., <test suite name>)` instead of `BOOST_AUTO_TEST_SUITE` and it should still work fine.  It might take a little bit of work to adapt it.

Reply:
I mean yes we can adapt it to catch but i don't know what purpose it serves ?

Reply:
Essentially this part of the code allows CTest integration for each model type.  So, with this code, we have the ability to have one CTest test suite for each model type, which is nice.  Otherwise, if you used CTest, it groups everything as one test suite, but especially for these models, tests for a model type may take a long time and so it might be preferable to have the ability to run only one test at a time.

Reply:
Makes absolute sense, I will implement it.